src/main/java/com/github/webicitybrowser/spec/fetch/imp/FetchEngineImp.java

```diff
@@ -42,8 +43,15 @@ private FetchResponse httpNetworkFetch(FetchParams params) {
 
 	private void fetchResponseHandover(FetchParams params, FetchResponse response) {
 		if (params.consumeBodyAction() != null) {
-			// TODO: Queue a response
+			if(response.body() == null) {
+				// TODO: Queue a response
+				fetchQueue.add(params.consumeBodyAction());
+			}
+			else {
+				params.consumeBodyAction().execute(response, true, new byte[]{});
```

Comment:
Can you make a method called queueAFetchTask (named after https://fetch.spec.whatwg.org/#queue-a-fetch-task)
For now, it can just directly queue a task onto the taskDestination